packages/coinstac-docker-manager/test/index.test.js

```diff
@@ -0,0 +1,348 @@
+  sinon.stub(docker, 'getImage').returns(getImageSuccessCallback);
+  const image = COMPUTATIONS[0].img;
+
+  await removeImage(image);
+  t.pass();
+
+  docker.getImage.restore();
+});
+
+test('setLogger', async (t) => {
+  const logger = winston.loggers.add('coinstac-main', {
+    level: 'silly',
+    transports: [new Console()],
+  });
+  setLogger(logger);
+
+  t.pass();
+});
+
+test('startService1', async (t) => {
+  sinon.stub(portscanner, 'findAPortNotInUse').returns(getRandomPort());
+  sinon.stub(docker, 'createContainer').returns({
+    start: sinon.stub(),
+    stop: sinon.stub(),
+    inspect: sinon.stub().yields(null, { State: { Running: true } }),
+  });
+
+  const opts = {
+    docker: { CMD: [] },
+  };
+
+  await startService('service-1', 'test-1', opts);
+  t.pass();
+
+  portscanner.findAPortNotInUse.restore();
+  docker.createContainer.restore();
+});
+
+test.serial.before('startService2', async (t) => {
+  sinon.stub(portscanner, 'findAPortNotInUse').returns(getRandomPort());
+  sinon.stub(docker, 'createContainer').returns({
+    start: sinon.stub(),
+    stop: sinon.stub(),
+    inspect: sinon.stub().yields('error', null),
+  });
+
+  const opts = {
+    docker: { CMD: [] },
+  };
+
+  await startService('service-2', 'test-2', opts);
+  t.pass();
+
+  portscanner.findAPortNotInUse.restore();
+  docker.createContainer.restore();
+});
+
+test.serial.before('startService3', async (t) => {
+  sinon.stub(portscanner, 'findAPortNotInUse').returns(getRandomPort());
+  sinon.stub(docker, 'createContainer').returns({
+    start: sinon.stub(),
+    stop: sinon.stub(),
+    inspect: sinon.stub().yields(null, { State: { Running: true } }),
+  });
+
+  const opts = {
+    docker: { CMD: [] },
+  };
+
+  await startService('service-3', 'test-3', opts);
+  await startService('service-3', 'test-3', opts);
+  t.pass();
+
+  portscanner.findAPortNotInUse.restore();
+  docker.createContainer.restore();
+});
+
+test.serial.before('startService4', async (t) => {
+  sinon.stub(portscanner, 'findAPortNotInUse').returns(getRandomPort());
+  sinon.stub(docker, 'createContainer').returns({
+    start: sinon.stub(),
+    stop: sinon.stub(),
+    inspect: sinon.stub().yields(null, { State: { Running: false } }),
+  });
+
+  const opts = {
+    docker: { CMD: [] },
+  };
+
+  await startService('service-4', 'test-4', opts);
+  t.pass();
+
+  portscanner.findAPortNotInUse.restore();
+  docker.createContainer.restore();
+});
+
+test.serial.before('startService5', async (t) => {
+  sinon.stub(portscanner, 'findAPortNotInUse').returns(WS_SERVER_PORT);
+  sinon.stub(docker, 'createContainer').returns({
+    start: sinon.stub(),
+    stop: sinon.stub(),
+    inspect: sinon.stub().yields(null, { State: { Running: true } }),
+  });
+
+  const opts = {
+    docker: { CMD: [] },
+  };
+
+  const service = await startService('service-5', 'test-5', opts);
+  await service(['node', '', '']);
+
+  t.pass();
```

Comment:
This looks like it should be actually interaction with the service, but I see no verification of that